Public/Sdk/Public/Managed/helpers.dsc

```diff
@@ -75,14 +78,15 @@ namespace Helpers {
 
         for (let ref of allReferences)
         {
-            computeTransitiveReferenceClosureHelper(ref, results, visitedReferences, compile);
+            const referencesToSkipSet = Set.empty<Deployment.DeployableItem>().add(...(referencesToSkip ? referencesToSkip : []));

```

Comment:
>nst referencesToSkipSet = S [](start = 14, length = 27)

I beleive in another place we have less recursino by simply exposing the the visited and toskip in one set.

i.e. you'd seed the visisted set with the set to skip

Reply:
Seeding the visited set with the content to skip elements is not enough, because the elements to skip are not only items with type Shared.Reference, but they could be assemblies, or have references themselves. So the content to skip needs to get decomposed into leaf elements, which is already what computeTransitiveReferenceClosureHelper is sort of doing

---
In reply to: [301267844](https://github.com/microsoft/BuildXL/pull/561#discussion_r301267844) [](ancestors = 301267844)